Detectors/EMCAL/workflow/src/RawToCellConverterSpec.cxx

```diff
@@ -105,29 +106,49 @@ void RawToCellConverterSpec::run(framework::ProcessingContext& ctx)
       std::cout << decoder.getRCUTrailer() << std::endl;
 
       o2::emcal::Mapper map = mMapper->getMappingForDDL(feeID);
+      int iSM = feeID / 2;
 
       // Loop over all the channels
       for (auto& chan : decoder.getChannels()) {
 
-        int iRow = map.getRow(chan.getHardwareAddress());
-        int iCol = map.getColumn(chan.getHardwareAddress());
-        ChannelType_t chantype = map.getChannelType(chan.getHardwareAddress());
-        int iSM = feeID / 2;
+        int iRow, iCol;
+        ChannelType_t chantype;
+        try {
+          iRow = map.getRow(chan.getHardwareAddress());
+          iCol = map.getColumn(chan.getHardwareAddress());
+          chantype = map.getChannelType(chan.getHardwareAddress());
+        } catch (Mapper::AddressNotFoundException& ex) {
+          std::cerr << ex.what() << std::endl;
+          continue;
+        };
 
         int CellID = mGeometry->GetAbsCellIdFromCellIndexes(iSM, iRow, iCol);
 
         // define the conatiner for the fit results, and perform the raw fitting using the stadnard raw fitter
-        o2::emcal::CaloFitResults fitResults = mRawFitter.evaluate(chan.getBunches(), 0, 0);
-
-        if (fitResults.getAmp() < 0 && fitResults.getTime() < 0) {
+        double amp = 0., time = 0.;
+        o2::emcal::CaloFitResults fitResults = mRawFitter->evaluate(chan.getBunches(), 0, 0);
+        if (fitResults.getAmp() > 0)
           fitResults.setAmp(0.);
+        if (fitResults.getTime() < 0)
           fitResults.setTime(0.);
-        }
-        mOutputCells.emplace_back(CellID, fitResults.getAmp(), fitResults.getTime(), chantype);
+        currentCellContainer->emplace_back(CellID, amp, time, chantype);
       }
     }
   }
 
+  // Loop over BCs, sort cells with increasing tower ID and write to output containers
+  mOutputCells.clear();
+  mOutputTriggerRecords.clear();
+  for (auto [bc, cells] : cellBuffer) {
+    mOutputTriggerRecords.emplace_back(bc, mOutputCells.size(), cells->size());
+    if (cells->size()) {
+      // Sort cells according to cell ID
+      std::sort(cells->begin(), cells->end(), [](o2::emcal::Cell& lhs, o2::emcal::Cell& rhs) { return lhs.getTower() < rhs.getTower(); });
+      for (auto cell : *cells)
+        mOutputCells.push_back(cell);
+    }
+  }
+
   LOG(DEBUG) << "[EMCALRawToCellConverter - run] Writing " << mOutputCells.size() << " cells ...";
   ctx.outputs().snapshot(o2::framework::Output{"EMC", "CELLS", 0, o2::framework::Lifetime::Timeframe}, mOutputCells);
   ctx.outputs().snapshot(o2::framework::Output{"EMC", "CELLSTRGR", 0, o2::framework::Lifetime::Timeframe}, mOutputTriggerRecords);
```

Comment:
If you send such a message with identical subspecs, there might be a conflict in the TF builder. I think you should use as a subspec something, which will be unique for given FLP, e.g. on of the FEEIDs. The subspec does not need to be meaningful, just unique.

Reply:
In the current approach we merge the cells from all links, therefore FEE ID is no longer well defined. In this case one would need to propagate the information which FLP is sending the data.

The situation could become simpler if we have one output spec per link (see above). Then we would just take the subspec ID as used in the STF builder.

Reply:
@mfasDa : I understand that you are merging data of all links on given FLP to single output. The problem is that with current code different FLPs will send to EPN messages with identical O2 headers, since you define output specs as ``{"EMC","CELLS",0}``. 
I am not sure how the ``TFBuilder`` will react on this (perhaps @ironMann can comment) but I think it is better to avoid this (won't cost anything). The subspec is just identifier, if it is N, it does not mean that preceding N-1 is expected.

I've referred to one of the FEEID used as a subspec for the FLP output just as an example of the entity which will make the different from other FLP (since the same FEEID is not supposed to be seen on different FLPs).